tests/components/madvr/test_config_flow.py

```diff
@@ -0,0 +1,90 @@
+"""Tests for the MadVR config flow."""
+
+from unittest.mock import AsyncMock
+
+from homeassistant.components.madvr.const import DEFAULT_NAME, DOMAIN
+from homeassistant.config_entries import SOURCE_USER
+from homeassistant.const import CONF_HOST, CONF_PORT
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResultType
+
+from .const import MOCK_CONFIG, MOCK_MAC
+
+from tests.common import MockConfigEntry
+
+
+async def test_full_flow(
+    hass: HomeAssistant,
+    mock_madvr_client: AsyncMock,
+    mock_setup_entry: AsyncMock,
+) -> None:
+    """Test full config flow."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}
+    )
+    assert result["type"] == FlowResultType.FORM
+    assert result["step_id"] == "user"
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        {CONF_HOST: MOCK_CONFIG[CONF_HOST], CONF_PORT: MOCK_CONFIG[CONF_PORT]},
+    )
+    await hass.async_block_till_done()
+
+    assert result["type"] == FlowResultType.CREATE_ENTRY
+    assert result["title"] == DEFAULT_NAME
+    assert result["data"] == {
+        CONF_HOST: MOCK_CONFIG[CONF_HOST],
+        CONF_PORT: MOCK_CONFIG[CONF_PORT],
+    }
+    assert result["result"].unique_id == MOCK_MAC
+    mock_madvr_client.open_connection.assert_called_once()
+    mock_madvr_client.async_add_tasks.assert_called_once()
+    mock_madvr_client.async_cancel_tasks.assert_called_once()
+
+
+async def test_flow_errors(
```

Comment:
Make sure the test ends in CREATE_ENTRY so we are sure we can recover from an error